Mirror ribs

chchatte92 opened the pull request:
### Briefly, what does this PR introduce?


### What kind of change does this PR introduce?
- [ ] Bug fix (issue #__)
- [ ] New feature (issue #__)
- [ ] Documentation update
- [ X] Other: __

### Please check if this PR fulfills the following:
- [ ] Tests for the changes have been added
- [ ] Documentation has been added / updated
- [X ] Changes have been communicated to collaborators

### Does this PR introduce breaking changes? What changes might users need to make to their code?
- No
### Does this PR change default behavior?
- No

chchatte92 commented:
This is a first attempt to add some ribs in the mirror! Similar ribs to be added to the aerogel. Currently we have C2F6 rib, we will make it with carbon-fiber.
![image](https://github.com/eic/epic/assets/52901208/264d8e98-95f1-4e18-aaf5-a718da44a30d)

wdconinc commented:
Use an updated base branch. Until you do, checks will fail.

chchatte92 commented:
Hi @wdconinc thanks a lot!  I just noticed it. I will take care of it.

c-dilks commented:
Is this PR ready for review?

c-dilks commented on src/DRICH_geo.cpp:
```diff
@@ -362,7 +362,9 @@ static Ref_t createDetector(Detector& desc, xml::Handle_t handle, SensitiveDetec
     double mirrorThetaRot = std::asin(mirrorCenterX / mirrorRadius);
     double mirrorTheta1   = mirrorThetaRot - std::asin((mirrorCenterX - mirrorRmin) / mirrorRadius);
-    double mirrorTheta2   = mirrorThetaRot + std::asin((mirrorRmax - mirrorCenterX) / mirrorRadius);
+    double mirrorTheta2   = 0.4*mirrorThetaRot + std::asin((mirrorRmax - mirrorCenterX) / mirrorRadius);
+    double mirrorTheta3   = 0.41*mirrorThetaRot + std::asin((mirrorRmax - mirrorCenterX) / mirrorRadius);
+    double mirrorTheta4   = 1.0*mirrorThetaRot + std::asin((mirrorRmax - mirrorCenterX) / mirrorRadius);
```
Move hard-coded numbers (0.4 and 0.41) to `drich.xml`.

c-dilks commented on src/DRICH_geo.cpp:
```diff
@@ -384,17 +388,31 @@ static Ref_t createDetector(Detector& desc, xml::Handle_t handle, SensitiveDetec
     // cut overlaps with other sectors using "pie slice" wedges, to the extent specified
     // by `mirrorPhiw`
     Tube              pieSlice(0.01 * cm, vesselRmax2, tankLength / 2.0, -mirrorPhiw / 2.0, mirrorPhiw / 2.0);
+    Tube              pieSlice1(0.01 * cm, vesselRmax2, tankLength / 2.0, -mirrorPhiw / 2.0, (0.05*mirrorPhiw) / 2.0);
+    Tube              pieSlice2(0.01 * cm, vesselRmax2, tankLength / 2.0, (0.06*mirrorPhiw) / 2.0, mirrorPhiw / 2.0);
     IntersectionSolid mirrorSolid2(pieSlice, mirrorSolid1, mirrorPlacement);
+    IntersectionSolid mirrorSolid4(pieSlice1, mirrorSolid3, mirrorPlacement);
+    IntersectionSolid mirrorSolid5(pieSlice2, mirrorSolid3, mirrorPlacement);
```
Perhaps it's time to come up with clearer names than using incrementing numbers, since we are now at "5".

c-dilks replied:
I don't have any good ideas, but maybe something like
- mirrorSolidInnerTile
- mirrorSolidOuterTileA
- mirrorSolidOuterTileB

Or you could put these in some STL container, such as `std::set`, so that you don't really have to give them names.

c-dilks commented on src/DRICH_geo.cpp:
```diff
@@ -384,17 +388,31 @@ static Ref_t createDetector(Detector& desc, xml::Handle_t handle, SensitiveDetec
 
     // mirror volume, attributes, and placement
-    Volume mirrorVol(detName + "_mirror_" + secName, mirrorSolid2, mirrorMat);
+    Volume mirrorVol(detName + "_mirror_0" + secName, mirrorSolid2, mirrorMat);
+    Volume mirrorVol2(detName + "_mirror_1" + secName, mirrorSolid4, mirrorMat);
+    Volume mirrorVol3(detName + "_mirror_2" + secName, mirrorSolid5, mirrorMat);
```
```suggestion
    Volume mirrorVol(detName + "_mirror_tile0" + secName, mirrorSolid2, mirrorMat);
    Volume mirrorVol2(detName + "_mirror_tile1" + secName, mirrorSolid4, mirrorMat);
    Volume mirrorVol3(detName + "_mirror_tile2" + secName, mirrorSolid5, mirrorMat);
```
or some suitable name, just so it's clear what `0`, `1`, and `2` mean.

chchatte92 commented:
@c-dilks I have changed the variables names to something more meaningful and have also added a set of ribs.

chchatte92 commented:
> @c-dilks I have changed the variables names to something more meaningful and have also added a set of ribs.

This is how it looks now:
![image](https://github.com/eic/epic/assets/52901208/f805e59a-5219-4e96-a114-b2b86723c98f)
The rib material is now Aluminium. I was wandering to make it in future to carbon fibre.

c-dilks commented:
Looks good enough to me.
- Are the optics the same?
- Does this supersede #380 ?

wdconinc reviewed (commented):
I'll defer to @c-dilks here

chchatte92 commented:
> I'll defer to @c-dilks here

@c-dilks hahahaha

c-dilks commented:
LGTM. Mark ready and add to merge queue.

chchatte92 commented:
Please don't merge now!

chchatte92 commented:
> LGTM. Mark ready and add to merge queue.

@c-dilks I had a discussion with Macro today, and he is right. That this gap should not be carbon fibre but C2F6, as these open region is useful for alignment of the mirror. This aligns with my first implementation. I will change the rib material.

rahmans1 commented:
@chchatte92 What is the status  of this PR? Should it be included in the February release for simulation campaign?
